bindsnet/pipeline/environment_pipeline.py

```diff
@@ -79,42 +74,38 @@ def __init__(
         self.reward_plot = None
 
         self.first = True
-        self.analyzer = MatplotlibAnalyzer()
+        self.analyzer = MatplotlibAnalyzer(**self.plot_config)
 
-    def init_fn(self):
+    def init_fn(self) -> None:
         pass
 
-    def train(self) -> None:
+    def train(self, **kwargs) -> None:
+        # language=rst
         """
-        Runs for the specified number of episodes from the Environment.
-        Each episode can be an arbitrary length.
+        Runs for the specified number of episodes. Each episode can be an arbitrary length.
         """
-
-        for self.episode in range(self.num_episodes):
+        while self.episode < self.num_episodes:
             self.reset_()
 
-            for step in itertools.count():
+            for _ in itertools.count():
                 batch = self.env_step()
```

Comment:
This should probably just be unpacked as `state, reward, done, info`, like usual.

Reply:
Wouldn't that be superfluous? It's easier and more clear (in my opinion) to move all these things between the many functions. Besides, I don't think it's done like that anywhere else.

Reply:
IMO, it makes the code more readable.

Reply:
Like this?